### PR TITLE
docs: improve AppMeta documentation

### PR DESCRIPTION
**Pull request by metaboulie**
## 📝 Summary

image

This PR improves the documentation for the `AppMeta` class and `app_meta()` function by adding clear examples and better explaining their functionality.

## 🔍 Description of Changes

- Added practical examples showing how to use `theme` with plotting libraries
- Added examples showing conditional content display based on `mode`
- Clarified return values for both `theme` and `mode` properties
- Ensured consistent documentation style with rest of codebase

## Examples Added
- Using `theme` to conditionally set Altair's theme
- Using `mode` to show/hide content based on execution context

## 📋 Checklist

- [x] I have read the [contributor guidelines](https://github.com/marimo-team/marimo/blob/main/CONTRIBUTING.md).
- [x] For large changes, or changes that affect the public API: this change was discussed or approved through an issue, on [Discord](https://marimo.io/discord?ref=pr), or the community [discussions](https://github.com/marimo-team/marimo/discussions) (Please provide a link if applicable).
- [ ] I have added tests for the changes made.
- [x] I have run the code and verified that it works as expected.

## 📜 Reviewers

@akshayka OR @mscolnick

### CODE REVIEW

File: marimo/_runtime/app_meta.py [hunk 1]
```diff
@@ -11,16 +11,42 @@ class AppMeta:
     """
     Metadata about the app.
 
-    This is used to store metadata about the app
-    that is not part of the app's code or state.
+    This class provides access to runtime metadata about a marimo app, such as
+    its display theme and execution mode. Get an instance with `marimo.app_meta()`.
```
Comment (mscolnick) [inline marimo/_runtime/app_meta.py:15]: nit: can we remove `Get an instance with `marimo.app_meta()`. and when used in `runtime.py`

File: marimo/_runtime/app_meta.py [hunk 2]
```diff
@@ -11,16 +11,42 @@ class AppMeta:
...
+
+    **Examples**:
```
Comment (mscolnick) [inline marimo/_runtime/app_meta.py:17]: can you move each example to `theme` and `mode` respectively.

good to keep as-is in `runtime.py`

**Comment by metaboulie** [comment]
the updated screenshot

image

File: marimo/_runtime/app_meta.py [hunk 3]
```diff
@@ -11,16 +11,37 @@ class AppMeta:
     """
     Metadata about the app.
 
-    This is used to store metadata about the app
-    that is not part of the app's code or state.
+    This class provides access to runtime metadata about a marimo app, such as
+    its display theme and execution mode. Get an instance with `marimo.app_meta()`.
```
Comment (mscolnick) [inline marimo/_runtime/app_meta.py:15]:
```suggestion
    its display theme and execution mode.
```

**Review by mscolnick — APPROVED** [review]

**Pull request merged**